src/View/Widget/DateTimeWidget.php

```diff
@@ -104,22 +107,25 @@ public function render(array $data, ContextInterface $context): string
             'templateVars' => [],
         ];
 
+        if (!isset($this->formatMap[$data['type']])) {
+            throw new InvalidArgumentException(sprintf(
+                'Invalid type "%s" for input tag',
```

Comment:
Could we include the list of valid types here? So its easier to figure out what a valid type is.

Reply:
Sure. This was just moved up from the function below though.